src/network/wait-online/wait-online.c

```diff
@@ -105,7 +110,9 @@ static int parse_argv(int argc, char *argv[]) {
         return 1;
 }
 
-int main(int argc, char *argv[]) {
+static int run(int argc, char *argv[]) {
+        _cleanup_(notify_on_exit) const char *notify_message_on_failure =
+                "STATUS=Failed waiting for network connectivity...";
```

Comment:
That's just too much magic, with the failure code pre-configured and overwritten on success.

Reply:
Yeah, I think this is tricky :-)